src/pybamm/dispatch/entry_points.py

```diff
@@ -109,7 +115,28 @@ def __getattribute__(self, name):
 models = EntryPoint(group="pybamm_models")
 
 
-def Model(model: str, *args, **kwargs):
+def get_cache_path(url):
+    cache_dir = Path.home() / ".pybamm_cache" / "pybamm" / "models"
+    cache_dir.mkdir(parents=True, exist_ok=True)
+    file_hash = hashlib.md5(url.encode()).hexdigest()
+    return cache_dir / f"{file_hash}.json"
+
+
+def clear_model_cache():
+    cache_dir = Path.home() / ".pybamm_cache" / "pybamm" / "models"
+    if cache_dir.exists():
+        for file in cache_dir.glob("*.json"):
+            file.unlink()
```

Comment:
This should be handled using `platformdirs.user_cache_dir`: https://platformdirs.readthedocs.io/en/latest/api.html#cache-directory